### PR TITLE
Added SAC agents

### PR DESCRIPTION
**Pull request by dapatil211**

### CODE REVIEW

File: hive/agents/qnets/sac_heads.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            )
+        ).sum(axis=1, keepdim=True)
+        log_prob = log_prob - correction
+        log_prob = log_prob.sum(axis=1, keepdim=True)
+        return action, log_prob, None
+
+
+class CategoricalPolicyHead(torch.nn.Module):
+    """A module that implements a discrete actor head. It uses the ouput from
+    the :obj:`actor_net`, and adds creates a
```
Comment (hnekoeiq) [inline hive/agents/qnets/sac_heads.py]: typo?

File: hive/configs/mujoco/sac.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    -
+      name: WandbLogger
+      kwargs:
+        project: Hive
+        name: *run_name
+        resume: "allow"
+        start_method: "fork"
+        tags: 
+          - benchmark_sac
+
```
Comment (hnekoeiq) [inline hive/configs/mujoco/sac.yml:71]: remove?

File: hive/agents/sac.py [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+        if self._training and not self._learn_schedule.get_value():
+            return (self._action_space.sample(), SACAgentState())
+
+        observation = torch.tensor(
+            np.expand_dims(observation, axis=0), device=self._device
+        ).float()
+        action, _, _ = self._actor(observation)
+        action = action.cpu().detach().numpy()
+        action = self.unscale_actions(action)
+        # action = action
```
Comment (hnekoeiq) [inline hive/agents/sac.py]: What this comment is for?

**Review by hnekoeiq — CHANGES_REQUESTED** [review]
Some minor comments. Otherwise, LGTM. Could you also add the link to the results?

**Comment by dapatil211** [comment]
Link to results comparing with CleanRL: https://api.wandb.ai/links/dapatil211/1df29fog

**Review by alirahkay — APPROVED** [review]
LGTM

**Review by hnekoeiq — APPROVED** [review]

**Pull request merged**